packages/dot-files/.cursor/rules/event-steps.mdc

```diff
@@ -1,114 +1,119 @@
 ---
-description: What are Motia Event Steps and examples using Typescript, Python and Ruby
+description: 
 globs: 
 alwaysApply: false
 ---
+
 # Event Steps Guide
 
-Event Steps handle asynchronous message processing and event emission.
-This is where you should put most of the business logic.
-Calling generative LLMs, accessing files, python or ruby logic.
+Event Steps handle asynchronous events and are the core building blocks of Motia workflows.
 
-## Examples
+## Basic Event Step Pattern
 ```typescript
 import { EventConfig, Handlers } from 'motia'
 import { z } from 'zod'
- 
+
 export const config: EventConfig = {
   type: 'event',
-  name: 'stepA',
-  description: 'Hello from Step A',
-  subscribes: ['pms.start'],
-  emits: ['pms.stepA.done'],
-  input: z.object({ message: z.string() }),
-  flows: ['parallel-merge'],
+  name: 'process-user',
+  description: 'Process new user registration',
+  subscribes: ['user.created'],
+  emits: ['user.processed'],
+  input: z.object({
+    name: z.string(),
+    email: z.string(),
+    id: z.string()
+  }),
+  flows: ['user-management']
 }
- 
-export const handler: Handlers['stepA'] = async (input, { emit, logger }) => {
-  logger.info('Processing message:', input.message)
- 
+
+export const handler: Handlers['process-user'] = async (input, { emit, logger, state }) => {
+  const { name, email, id } = input
+  
+  logger.info('Processing user', { id, name })
+  
+  // Your business logic here
+  const processedUser = {
+    ...input,
+    status: 'active',
+    createdAt: new Date().toISOString()
+  }
+  
+  // Store in state if needed
+  await state.set('user', processedUser)
```

Comment:
we're missing one level here

it's actually two levels, a group, and an item in a group.


```suggestion
  // Store in state if needed
  await state.set(traceId, 'user', processedUser)
```

a group is typically a top-level object, the item is usually a field in it. we can group by traceId or other things